Sources/Apodini/Request/DynamicProperty.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  DynamicProperty.swift
+//  
+//
+//  Created by Max Obermeier on 09.12.20.
+//
+
+import Foundation
+import Runtime
+
+/// `DynamicProperty` allows for wrapping Apodini's property wrappers while maintianing their functionality. By
+/// conforming a `struct` to `DynamicProperty` you this `struct`'s properties discoverable to the Apodini
+/// runtime framework. This can be used to e.g. compine two property wrappers provided by the Apodini framework
+/// into one that combines their functionality
+/// - Precondition: Only structs can be a DynamicProperty
+public protocol DynamicProperty { }
```

Comment:
If we have to keep this protocol around I would suggest to rename it to `DynamicParameter`.  While it might use some other property wrappers such as `@Environment` or `@ObservedObjects` I see that it behaves like a complex `@Parameter` for the implementations that developers using Apodini write. As we do not have the concept of a property in Apodini but `Parameter`s I would think that this would be a more descriptive name.
Alternatively I would image that `@Parameter` conforms to `Injectable` which is public and we use an internal protocol `_Injectable` to implement the inject functionality for the internal implementations that also include `execute` and `apply`.

Reply:
Again: `DynamicProperty` has nothing to do with `@Parameter` in the first place. A user of Apodini could combine a `@Environment(\.database)` and an `@ObservedObject` (publishing user-ids)  into a new property wrapper which publishes the complete user-object as fetched from the database.